feature/notification/impl/src/commonMain/kotlin/net/thunderbird/feature/notification/impl/DefaultNotificationRegistry.kt

```diff
@@ -3,35 +3,48 @@ package net.thunderbird.feature.notification.impl
 import kotlin.concurrent.atomics.AtomicInt
 import kotlin.concurrent.atomics.ExperimentalAtomicApi
 import kotlin.concurrent.atomics.incrementAndFetch
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.sync.Mutex
 import kotlinx.coroutines.sync.withLock
 import net.thunderbird.feature.notification.api.NotificationId
 import net.thunderbird.feature.notification.api.NotificationRegistry
 import net.thunderbird.feature.notification.api.content.Notification
 
 @OptIn(ExperimentalAtomicApi::class)
-class DefaultNotificationRegistry : NotificationRegistry {
+class DefaultNotificationRegistry(
+    dispatcher: CoroutineDispatcher = Dispatchers.Main,
+) : NotificationRegistry {
+    private val scope: CoroutineScope = CoroutineScope(dispatcher)
```

Comment:
I think this should use a `SupervisorJob` to avoid leaking the scope.


```suggestion
    private val scope: CoroutineScope = CoroutineScope(SupervisorJob() + dispatcher)
```